### PR TITLE
[ENH] Kalman filtering gps

### PR DESCRIPTION
**Pull request by rksh**
adding optional kalman filtering

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java [hunk 1]
```diff
@@ -86,7 +86,7 @@
 import static net.wigle.wigleandroid.listener.GPSListener.MIN_ROUTE_LOCATION_PRECISION_METERS;
 
 /**
- * show a map depicting current position and configurable stumbling progress information.f
+ * show a map depicting current position and configurable stumbling progress information.
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MappingFragment.java:89]: This was a load-bearing 'f', now the entire codebase has collapsed. Hope you're happy.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java [hunk 2]
```diff
@@ -170,6 +176,19 @@ private void updateLocationData( final Location newLocation ) {
...
 
         boolean newOK = newLocation != null;
+        if (null != newLocation && null != kalmanLatLong &&  kalmanLatLong.getAccuracy() < 0) {
+            kalmanLatLong.setState(newLocation.getLatitude(), newLocation.getLongitude(), newLocation.getAccuracy(), newLocation.getTime());
+        } else if (null != newLocation) {
+            if (null != kalmanLatLong) {
+                kalmanLatLong.process(newLocation.getLatitude(), newLocation.getLongitude(), newLocation.getAccuracy(), newLocation.getTime());
+                MainActivity.error("KALMAN TEST: [lat: " + newLocation.getLatitude() + " v. (k):" +
+                        kalmanLatLong.getLat() + " lon: " + newLocation.getLongitude() + " v. (k):" +
+                        kalmanLatLong.getLng() + "], acc(k): " + kalmanLatLong.getAccuracy() + " location prov: " + newLocation.getProvider());
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:186]: Maybe don't want this logging errors?

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:186]: argh, yeah, left my debug in.

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/listener/GPSListener.java:186]: ...aaand the fact that this wasn't triggering in my testing points out another careless mistake that invalidates all my testing. pushing fix, but setting "do not merge" pending testing.

**Comment by rksh** [comment]
clear to merge.

**Pull request merged**